diff --git a/.npmignore b/.npmignore
deleted file mode 100644
index 1ca957177f035203810612d1d93a66b08caff296..0000000000000000000000000000000000000000
diff --git a/lib/deceiver.js b/lib/deceiver.js
index 33e14cbd6ed4f07df56f031676600b5fc1941344..fd9e6f6af7250edf552e0e2dc586d18b627c1dda 100644
--- a/lib/deceiver.js
+++ b/lib/deceiver.js
@@ -10,6 +10,17 @@ var mode = /^v0\.8\./.test(process.version) ? 'rusty' :
            'modern';
 
 var HTTPParser;
+var httpParserBinding;
+
+try {
+  httpParserBinding = process.binding('http_parser');
+} catch (e) {
+  httpParserBinding = null;
+}
+
+if (!httpParserBinding && (mode === 'normal' || mode === 'modern')) {
+  mode = 'old';
+}
 
 var methods;
 var reverseMethods;
@@ -19,12 +30,12 @@ var kOnHeadersComplete;
 var kOnMessageComplete;
 var kOnBody;
 if (mode === 'normal' || mode === 'modern') {
-  HTTPParser = process.binding('http_parser').HTTPParser;
+  HTTPParser = httpParserBinding.HTTPParser;
   methods = HTTPParser.methods;
 
   // v6
   if (!methods)
-    methods = process.binding('http_parser').methods;
+    methods = httpParserBinding.methods;
 
   reverseMethods = {};
 
